solarforecastarbiter/reference_forecasts/main.py

```diff
@@ -473,19 +477,75 @@ def make_latest_nwp_forecasts(token, run_time, issue_buffer, base_url=None):
     base_url : str or None, default None
         Alternate base_url of the API
     """
+    session, forecast_df = _get_nwp_forecast_df(token, run_time, base_url)
+    execute_for = forecast_df[
+        forecast_df.next_issue_time <= run_time + issue_buffer]
+    if execute_for.empty:
+        logger.info('No forecasts to be made at %s', run_time)
+        return
+    process_nwp_forecast_groups(session, run_time, execute_for)
+
+
+def _get_nwp_forecast_df(token, run_time, base_url):
     session = api.APISession(token, base_url=base_url)
     user_info = session.get_user_info()
     forecasts = session.list_forecasts()
     forecasts += session.list_probabilistic_forecasts()
     forecasts = [fx for fx in forecasts
                  if fx.provider == user_info['organization']]
     forecast_df = find_reference_nwp_forecasts(forecasts, run_time)
-    execute_for = forecast_df[
-        forecast_df.next_issue_time <= run_time + issue_buffer]
-    if execute_for.empty:
-        logger.info('No forecasts to be made at %s', run_time)
-        return
-    process_nwp_forecast_groups(session, run_time, execute_for)
+    return session, forecast_df
+
+
+def _nwp_issue_time_generator(fx, last_timestamp, next_timestamp):
```

Comment:
I feel like I understand what this function is doing and how it's used in the calling loop, but I don't understand the parameter names. Nor do I understand the comments. So I'm not confident I really know what's going on.

I thought the parameter names would be `gap_start` and `gap_end`. I guess that's similar to `last_existing_timestamp` and `next_existing_timestamp` and if you've taken out "existing" for brevity then we're on the same page.

Reply:
yes, took out "existing". modified the parameters and expanded comments a bit